Providers/Profile.swift

```diff
@@ -39,7 +39,15 @@ public protocol SyncManager {
     func onAddedAccount() -> Success
 }
 
-typealias SyncFunction = (SyncDelegate, Prefs, Ready, OldSyncReason) -> OldSyncResult
+/**
+ * This exists to pass in external context: e.g., the UIApplication can
+ * expose notification functionality in this way.
+ */
```

Comment:
nit, normally prefer comment format with `///`. This is also the provided comment with shortcut `option` + `cmd` + `/` when the cursor is on top of a function in Xcode.
```suggestion
/// This exists to pass in external context: e.g., the UIApplication can
/// expose notification functionality in this way.
```